.github/workflows/main.yml

```diff
@@ -1,23 +1,53 @@
 # This is a basic workflow to help you get started with Actions
 
-name: XGoost-CI
+name: XGBoost-CI
 
 # Controls when the action will run. Triggers the workflow on push or pull request
 # events but only for the master branch
 on: [push, pull_request]
 
 # A workflow run is made up of one or more jobs that can run sequentially or in parallel
 jobs:
+  test-with-jvm:
+    name: Test JVM on OS ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [windows-latest, windows-2016, ubuntu-latest]
+
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        submodules: 'true'
+
+    - uses: actions/setup-java@v1
+      with:
+        java-version: 1.8
+
+    - name: Test JVM packages
+      run: |
+        cd jvm-packages
+        mvn test -pl :xgboost4j_2.12
+
+
   test-with-R:
     runs-on: ${{ matrix.config.os }}
 
-    name: Test R on OS ${{ matrix.config.os }}, R (${{ matrix.config.r }})
+    name: Test R on OS ${{ matrix.config.os }}, R ${{ matrix.config.r }}, Compiler ${{ matrix.config.compiler }}, Build ${{ matrix.config.build }}
```

Comment:
Quite nice of the R community to publish GitHub Actions like `r-lib/actions/setup-r`. It would have been quite painful to set everything up otherwise.